apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-preview-dropdown-selected.js

```diff
@@ -0,0 +1,30 @@
+import { __ } from '@wordpress/i18n';
+import tracksRecordEvent from './track-record-event';
+
+/**
+ * Return the event definition object to track `wpcom_block_editor_template_part_detach_blocks`.
+ * @returns {import('./types').DelegateEventHandler} event object definition.
+ */
+export default () => ( {
+	id: 'wpcom-preview-dropdown-selected',
+	selector: '.components-menu-item__button',
+	type: 'click',
+	capture: true,
+	handler: ( _event, target ) => {
+		const mapTextToPreviewMode = {
+			[ __( 'Desktop' ) ]: 'desktop',
```

Comment:
Does this re-use the same translations as Gutenberg? If not, and if wpcom translators use a different wording than Gutenberg translators, I guess we might be unable to track some preview modes in other languages.

Before tracking any event, could we check whether every preview mode hardcoded here exists in the dropdown? 

I want to avoid something like the following:

English | Spanish (Gutenberg) | Spanish (wpcom) | Same translation?
--- | --- | --- | ---
Desktop | Escritorio | Escritorio | ✅
Desktop (50%) | Escritorio (50%) | Escritorio (50%) | ✅
Tablet | Tableta | Tablet | ❌ (*)
Mobile | Móvil | Móvil | ✅

_(*) In Spanish we can say both Tablet and Tableta._

We would be missing the clicks on the Tableta option (because the logic here expects a Tablet option) which could lead to a wrong data analysis. To prevent that, I think we should check that all preview modes we're expecting exist:

```js
const shouldTrack = mapTextToPreviewMode.every( ( previewMode ) => /* Check if mode exists in dropdown*/ );
if ( shouldTrack ) {
  tracksRecordEvent( /* ... */ );
}
```

Reply:
Good question! I checked the translation and they're the same. Let me think whether there is a better way to compare the text.

| Gutenberg | wpcom |
| - | - |
| <img width="249" alt="image" src="https://github.com/user-attachments/assets/ed6a3c18-19e5-463b-8879-86198743e6fd"> | <img width="342" alt="image" src="https://github.com/user-attachments/assets/8accd99b-45ae-4cf0-8f01-e34ba706bc82"> |

Reply:
Thank you! I might be overthinking here though – translations will likely match 99% of the times. Feel free to leave things as they are now if you struggle to find a better way!

Reply:
I assume the translation is always the same as I checked `__` used on both places equals to `window.wp.i18n.__`

<img width="187" alt="image" src="https://github.com/user-attachments/assets/f14229c5-f436-4d8f-a24f-fccdef5e699c">